tests/test_baggage.py

```diff
@@ -0,0 +1,161 @@
+    """
+    if isinstance(request_headers, dict):
+        # Case-insensitive lookup for baggage header since dependening on the weblog app implementation
+        for key, value in request_headers.items():
+            if key.lower() == "baggage":
+                return value
+    elif isinstance(request_headers, list):
+        for header in request_headers:
+            if header.get("key", "").lower() == "baggage":
+                return header.get("value")
+    return None
+
+
+@scenarios.default
+@features.datadog_baggage_headers
+class Test_Baggage_Headers_Basic:
+    def setup_main(self):
+        self.r = weblog.get(
+            "/make_distant_call",
+            params={"url": "http://weblog:7777"},
+            headers={"x-datadog-parent-id": "10", "x-datadog-trace-id": "2", "baggage": "foo=bar"},
+        )
+
+    def test_main(self):
+        interfaces.library.assert_trace_exists(self.r)
+        assert self.r.status_code == 200
+        data = json.loads(self.r.text)
+        baggage_value = extract_baggage_value(data["request_headers"])
+        assert baggage_value is not None
+        assert "foo=bar" in baggage_value
+
+
+@scenarios.default
+@features.datadog_baggage_headers
+class Test_Baggage_Headers_Malformed:
+    def setup_main(self):
+        self.r = weblog.get(
+            "/make_distant_call",
+            params={"url": "http://weblog:7777"},
+            headers={
+                "x-datadog-parent-id": "10",
+                "x-datadog-trace-id": "2",
+                "baggage": "no-equal-sign,foo=gets-dropped-because-previous-pair-is-malformed",
+            },
+        )
+
+    def test_main(self):
+        interfaces.library.assert_trace_exists(self.r)
+        assert self.r.status_code == 200
+        data = json.loads(self.r.text)
+        headers = data["request_headers"]
+        # When headers is a dict, ensure "baggage" key is absent.
+        if isinstance(headers, dict):
+            assert "baggage" not in headers
+        # When headers is a list, ensure no header entry has key "baggage"
+        elif isinstance(headers, list):
+            assert all(header.get("key") != "baggage" for header in headers)
+
+
+@scenarios.default
+@features.datadog_baggage_headers
+class Test_Baggage_Headers_Malformed2:
+    def setup_main(self):
+        self.r = weblog.get(
+            "/make_distant_call",
+            params={"url": "http://weblog:7777"},
+            headers={"x-datadog-parent-id": "10", "x-datadog-trace-id": "2", "baggage": "=no-key"},
+        )
+
+    def test_main(self):
+        interfaces.library.assert_trace_exists(self.r)
+        assert self.r.status_code == 200
+        data = json.loads(self.r.text)
+        headers = data["request_headers"]
+        if isinstance(headers, dict):
+            assert "baggage" not in headers
+        elif isinstance(headers, list):
+            assert all(header.get("key") != "baggage" for header in headers)
+
+
+@scenarios.only_baggage_propagation
```

Comment:
Do we need a separate scenario to only test the baggage propagator? I suspect that only having the `baggage` header in the incoming request (like you have here) would be sufficient, unless there are some edge cases that you had in mind